regex/tests/Test.lean

```diff
@@ -153,4 +153,907 @@ def word_class := Regex.parse! r##"\w+"##
+-- when Unicode is enabled, even when UTF-8 mode is disabled.
+
+-- FIXME: This test is not working as expected.
+-- -- name = "unicode5-not-noutf8"
+-- #eval unicode_non_word_boundary_02.captureAll "0\xFF\xFF\xFF\xFF"
+-- #guard unicode_non_word_boundary_02.captureAll "0\xFF\xFF\xFF\xFF" = #[]
+-- #guard unicode_non_word_boundary_02.bt.captureAll "0\xFF\xFF\xFF\xFF" = #[]
+
+-- But this DOES get matches since \B in ASCII mode only looks at individual
+-- bytes.
+
+-- FIXME: This test is not working as expected.
+-- name = "unicode5-not-noutf8-only-ascii"
+-- #eval unicode_non_word_boundary_02.captureAll "0\xFF\xFF\xFF\xFF"
+-- #guard unicode_non_word_boundary_02.captureAll "0\xFF\xFF\xFF\xFF" = #[
+--   ⟨"0\xFF\xFF\xFF\xFF", #[.some ⟨2⟩, .some ⟨2⟩]⟩,
+--   ⟨"0\xFF\xFF\xFF\xFF", #[.some ⟨3⟩, .some ⟨3⟩]⟩,
+--   ⟨"0\xFF\xFF\xFF\xFF", #[.some ⟨4⟩, .some ⟨4⟩]⟩,
+--   ⟨"0\xFF\xFF\xFF\xFF", #[.some ⟨5⟩, .some ⟨5⟩]⟩
+-- ]
+-- #guard unicode_non_word_boundary_02.bt.captureAll "0\xFF\xFF\xFF\xFF" = #[
+--   ⟨"0\xFF\xFF\xFF\xFF", #[.some ⟨2⟩, .some ⟨2⟩]⟩,
+--   ⟨"0\xFF\xFF\xFF\xFF", #[.some ⟨3⟩, .some ⟨3⟩]⟩,
+--   ⟨"0\xFF\xFF\xFF\xFF", #[.some ⟨4⟩, .some ⟨4⟩]⟩,
+--   ⟨"0\xFF\xFF\xFF\xFF", #[.some ⟨5⟩, .some ⟨5⟩]⟩
+-- ]
+
+-- Some tests of no particular significance.
+def unicode_word_boundary_03 := Regex.parse! r##"\b[0-9]+\b"##
+
+-- name = "unicode6"
+#guard unicode_word_boundary_03.captureAll "foo 123 bar 456 quux 789" = #[
+  ⟨"foo 123 bar 456 quux 789", #[.some ⟨4⟩, .some ⟨7⟩]⟩,
+  ⟨"foo 123 bar 456 quux 789", #[.some ⟨12⟩, .some ⟨15⟩]⟩,
+  ⟨"foo 123 bar 456 quux 789", #[.some ⟨21⟩, .some ⟨24⟩]⟩
+]
+#guard unicode_word_boundary_03.bt.captureAll "foo 123 bar 456 quux 789" = #[
+  ⟨"foo 123 bar 456 quux 789", #[.some ⟨4⟩, .some ⟨7⟩]⟩,
+  ⟨"foo 123 bar 456 quux 789", #[.some ⟨12⟩, .some ⟨15⟩]⟩,
+  ⟨"foo 123 bar 456 quux 789", #[.some ⟨21⟩, .some ⟨24⟩]⟩
+]
+
+-- name = "unicode7"
+#guard unicode_word_boundary_03.captureAll "foo 123 bar a456 quux 789" = #[
+  ⟨"foo 123 bar a456 quux 789", #[.some ⟨4⟩, .some ⟨7⟩]⟩,
+  ⟨"foo 123 bar a456 quux 789", #[.some ⟨22⟩, .some ⟨25⟩]⟩
+]
+#guard unicode_word_boundary_03.bt.captureAll "foo 123 bar a456 quux 789" = #[
+  ⟨"foo 123 bar a456 quux 789", #[.some ⟨4⟩, .some ⟨7⟩]⟩,
+  ⟨"foo 123 bar a456 quux 789", #[.some ⟨22⟩, .some ⟨25⟩]⟩
+]
+
+-- name = "unicode8"
+#guard unicode_word_boundary_03.captureAll "foo 123 bar 456a quux 789" = #[
+  ⟨"foo 123 bar 456a quux 789", #[.some ⟨4⟩, .some ⟨7⟩]⟩,
+  ⟨"foo 123 bar 456a quux 789", #[.some ⟨22⟩, .some ⟨25⟩]⟩
+]
+#guard unicode_word_boundary_03.bt.captureAll "foo 123 bar 456a quux 789" = #[
+  ⟨"foo 123 bar 456a quux 789", #[.some ⟨4⟩, .some ⟨7⟩]⟩,
+  ⟨"foo 123 bar 456a quux 789", #[.some ⟨22⟩, .some ⟨25⟩]⟩
+]
+
+-- A variant of the problem described here:
+-- https://github.com/google/re2/blob/89567f5de5b23bb5ad0c26cbafc10bdc7389d1fa/re2/dfa.cc#L658-L667
+
+def unicode_word_boundary_04 := Regex.parse! r##"(?:\b|%)+"##
+
+-- FIXME: This test is not working as expected.
```

Comment:
This failure is actually interesting and worth investigation. I opened #103 for investigation. Thank you!